### PR TITLE
Enhance wording + add warnings for default destinations

### PR DESCRIPTION
**Pull request by kgroschoff**
### What this PR does / why we need it
Today during the day we noticed, that the default destinations need some more warnings and the wording could be optimised.

### Special notes for your reviewer

If a destination is marked as default already and user wants to unset the checkbox following warning will appear:
delete-warning

If a user wants to select a new destination as default (no matter if via add or edit dialog), following warning will appear:
overwrite-warning

### Release note

```release-note
NONE
```

### CODE REVIEW

File: src/app/settings/admin/bucket-settings/destinations/destination-dialog/component.ts [hunk 1]
```diff
@@ -124,4 +134,8 @@ export class DestinationDialog implements OnInit, OnDestroy {
         return 'Destination was successfully edited';
     }
   }
+
+  private _isDefault(): boolean {
+    return !!this.currentDefault && this.currentDefault === this.data?.destination?.destinationName;
```
Comment (maciaszczykm) [inline src/app/settings/admin/bucket-settings/destinations/destination-dialog/component.ts:139]: Is `!!this.currentDefault` really needed?

**Comment by kgroschoff** [comment]
@maciaszczykm can you lgtm again? there was a merge conflict

**Pull request merged**